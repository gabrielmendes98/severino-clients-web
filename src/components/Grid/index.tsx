import MuiGrid, { GridProps as MuiGridProps } from '@mui/material/Grid';
import { ElementType } from 'react';

interface GridProps extends MuiGridProps {
  component?: ElementType<any> & (ElementType<any> | undefined);
}

const Grid = ({ children, component = 'div', ...props }: GridProps) => (
  <MuiGrid component={component} {...props}>
    {children}
  </MuiGrid>
);

export default Grid;
