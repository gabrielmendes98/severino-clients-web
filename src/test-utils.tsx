import React, { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import ThemeProvider from './common/providers/Theme';
import { AppStore, RootState, setupStore } from './common/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const customRender = (
  ui: ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) => {
  const Wrapper: FC<{ children: ReactNode }> = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider>{children}</ThemeProvider>
    </Provider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

/* eslint-disable */
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
/* eslint-enable */
