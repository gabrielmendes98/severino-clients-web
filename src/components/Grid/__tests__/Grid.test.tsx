import { render } from 'test-utils';
import Grid from '../index';

it('should match snapshot', () => {
  const { container } = render(
    <Grid container>
      <Grid item xs={6}>
        item 1
      </Grid>
      <Grid item xs={6}>
        item 2
      </Grid>
      <Grid item xs={4}>
        item 3
      </Grid>
      <Grid item xs={4}>
        item 4
      </Grid>
      <Grid item xs={4}>
        item 5
      </Grid>
    </Grid>,
  );
  expect(container).toMatchSnapshot();
});
