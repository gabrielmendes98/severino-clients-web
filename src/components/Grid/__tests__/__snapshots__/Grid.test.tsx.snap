// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      item 1
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      item 2
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      item 3
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      item 4
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      item 5
    </div>
  </div>
</div>
`;
